### ISSUE TITLE
Fix S2259 FN: Rule does not raise on LINQ methods

### ISSUE THREAD

**Issue by Tim-Pohlmann** [SonarSource/sonar-dotnet#9499]
```csharp
void Method(List<int> list)
{
    _ = list.FirstOrDefault();  // branching
    _ = list.Count();  // FN
}
```
S2259 should re-use the list of known LINQ methods introduced in #9414 to raise in cases like this.
